### PR TITLE
[JAX] Update jax_scaled_masked_softmax to match TE kernel implementation

### PR DESCRIPTION
**Pull request by jberchtold-nvidia**
# Description

Our pure-JAX implementation of masked scaled softmax differed from our TE kernel implementation. This PR updates `jax_scaled_masked_softmax` to match our TE kernel.

## Type of change

- [x] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Infra/Build change
- [x] Code refactoring

## Changes

* Update `jax_scaled_masked_softmax` to use `jax.nn.softmax(where=mask)` instead of subtracting large values from logits and update mask to `mask != 1` instead of `mask > 0` to match TE kernel implementation.

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by phu0ngng — APPROVED** [review]
LGTM.

**Comment by phu0ngng** [comment]
Pipeline #29075787.
